steps/virtual-device-testing-for-android/0.9.1/step.yml

```diff
@@ -0,0 +1,165 @@
+description: Run Android UI tests on virtual devices.
+website: https://github.com/bitrise-steplib/steps-virtual-device-testing-for-android
+source_code_url: https://github.com/bitrise-steplib/steps-virtual-device-testing-for-android
+support_url: https://github.com/bitrise-steplib/steps-virtual-device-testing-for-android/issues
+published_at: 2017-08-28T16:53:27.364922849+02:00
+source:
+  git: https://github.com/bitrise-steplib/steps-virtual-device-testing-for-android.git
+  commit: 11af5d56f7bce84c9642738c7b920b5c1be4f6e9
+host_os_tags:
+- ubuntu
+project_type_tags:
+- android
+type_tags:
+- test
+toolkit:
+  go:
+    package_name: github.com/bitrise-steplib/steps-virtual-device-testing-for-android
+deps:
+  brew:
+  - name: go
+  apt_get:
+  - name: golang
+    bin_name: go
+is_always_run: false
+is_skippable: false
+inputs:
+- apk_path: $BITRISE_APK_PATH
+  opts:
+    description: |
+      The path to the unaligned debug APK.
+    is_required: true
+    summary: The path to the debug, unaligned APK.
+    title: APK path
+- opts:
+    description: |
+      One device configuration per line and the parameters are separated with "," in the order of: deviceID, version, language, orientation
+      For example:
+      `NexusLowRes,24,en,portrait`
+      `NexusLowRes,24,en,landscape`
+    is_required: true
+    summary: |
+      One device configuration per line and the parameters are separated with "," in the order of: "deviceID, version, language, orientation". For example: `NexusLowRes,24,en,portrait`
+    title: Test devices
+  test_devices: NexusLowRes,24,en,portrait
+- opts:
+    description: |
+      The type of your test you want to run on the device.
+    is_required: true
+    summary: The type of your test
+    title: Test type
+    value_options:
+    - instrumentation
+    - robo
+    - gameloop
+  test_type: instrumentation
+- app_package_id: null
+  opts:
+    description: |
+      The package ID of your app
+    summary: The package ID of your apk
+    title: App package ID
+- opts:
+    category: Instrumentation Test
+    description: |
+      The path to the unaligned test debug APK.
+    summary: The path to the unaligned test debug APK.
+    title: Test APK path
+  test_apk_path: null
+- inst_test_package_id: null
+  opts:
+    category: Instrumentation Test
+    description: |
+      The package ID of your test apk
+    summary: The package ID of your test apk
+    title: Test package ID
+- inst_test_runner_class: null
+  opts:
+    category: Instrumentation Test
+    description: |
+      The test runner that will run your test cases
+    summary: The test runner that will run your test cases
+    title: Test runner class
+- inst_test_targets: null
+  opts:
+    category: Instrumentation Test
+    description: |
+      Test targets
+    summary: Test targets
+    title: |
+      Test targets, seperated with the "," character.
+- opts:
+    category: Robo Test
+    title: Initial activity
```

Comment:
this can be fetched from the apk (`aapt dump badging apk_path` - `launchable-activity`)

Reply:
if there are multiple activities in the apk, we cannot decide automatically which one to use, however user can override the default one